components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/storm/StormTopologyManager.java

```diff
@@ -206,16 +195,16 @@ public void run() {
                             Nimbus.Client client = NimbusClient.getConfiguredClient(stormConfig).getClient();
                             client.killTopologyWithOpts(topologyName, options);
                             waitForTopologyToBeRemoved(jobPrefix);
-                        } catch (TException e) {
+                        } catch (NotAliveException e) {
+                            log.info(jobPrefix + "Topology '" + topologyName + "' is not alive to kill");
```

Comment:
Here what we actually want is to kill the topology if it's already there to re-submit it. For some reason if isTopologyExisits give a false positive answer then when we try to kill it will give this exception. So, if the topology is not really there, that's what we want to happen. Therefore, it's not an erroneous condition.